packages/flutter_driver/lib/src/driver/refresh_rate_summarizer.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'timeline.dart';
+
+/// Event name for refresh rate related timeline events.
+const String kUIThreadVsyncProcessEvent = 'VsyncProcessCallback';
+
+/// Summarizes [TimelineEvents]s corresponding to `kUIThreadVsyncProcessEvent` events.
+///
+/// `RefreshRate` is the time between the start of a vsync pulse and the target time of that vsync.
+class RefreshRateSummarizer {
+  /// Creates a [RefreshRateSummarizer] given the timeline events.
+  RefreshRateSummarizer(this.vsyncEvents);
+
+  /// Timeline events with name: [kUIThreadVsyncProcessEvent].
+  final List<TimelineEvent> vsyncEvents;
+
+  /// Computes a [RefreshRateSummary] over the period of the timeline.
+  RefreshRateSummary computeRefreshRateSummary() {
+    return RefreshRateSummary._(refreshRates: _computeRefreshRates());
+  }
+
+  List<double> _computeRefreshRates() {
+    final List<double> result = <double>[];
+    for (int i = 0; i < vsyncEvents.length; i++) {
+      final TimelineEvent event = vsyncEvents[i];
+      if (event.phase != 'B') {
+        continue;
+      }
+      assert(event.name == kUIThreadVsyncProcessEvent);
+      assert(event.arguments != null);
+      final Map<String, dynamic> arguments = event.arguments!;
+      const double nanosecondsPerSecond = 1e+9;
+      final int startTimeInNanoseconds = arguments['StartTime'] as int;
+      final int targetTimeInNanoseconds= arguments['TargetTime'] as int;
+      final int frameDurationInNanoseconds = targetTimeInNanoseconds - startTimeInNanoseconds;
+      final double refreshRate = nanosecondsPerSecond /
+          frameDurationInNanoseconds;
+      result.add(refreshRate);
+    }
+    return result;
+  }
+}
```

Comment:
This class exists just to run one function on the data. Could this be just turned into a function, say on the summary class?

Reply:
good idea, done